python_modules/dagster/dagster/core/definitions/job_definition.py

```diff
@@ -320,15 +317,25 @@ def _get_graph_definition(
 
         # rebuild graph if any nodes inside the graph are selected
         if node.is_graph and resolved_op_selection_dict[node.name] is not LeafNodeSelection:
-            definition = _get_graph_definition(
+            sub_definition, sub_ignored_solids_dict = _get_graph_definition(
                 node.definition,
                 resolved_op_selection_dict[node.name],
                 parent_handle=node_handle,
             )
+            ignored_solids_dict[node.name] = sub_ignored_solids_dict
+            # build a new node using the subsetted definition
+            selected_node = Node(
+                name=node.name,
+                definition=sub_definition,
+                graph_definition=node.graph_definition,
+                tags=node.tags,
+                hook_defs=node.hook_defs,
+                retry_policy=node.retry_policy,
+            )
```

Comment:
[1] call-sites like this terrify me since if we add a new property on `Node` this transfer could miss it and tracking down the subtle bug would be nightmarish